packages/slice-machine/src/features/customTypes/customTypesBuilder/UIDEditor.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActionButton,
+  DialogActions,
+  DialogCancelButton,
+  DialogContent,
+  DialogHeader,
+  Form,
+  FormInput,
+  Text,
+} from "@prismicio/editor-ui";
+import { useState } from "react";
+
+import {
+  addUIDField,
+  getFieldLabel,
+  getUIDFieldEntry,
+  updateUIDField,
+} from "@/domain/customType";
+import { UIDFieldCustomErrorMap, UIDFieldLabelSchema } from "@/domain/fields";
+import { useCustomTypeState } from "@/features/customTypes/customTypesBuilder/CustomTypeProvider";
+
+export function UIDEditor() {
+  const [isOpen, setOpen] = useState(false);
+  const { customType, setCustomType } = useCustomTypeState();
+  const [_, field] = getUIDFieldEntry(customType) ?? [];
+  const uidFieldLabel = field ? getFieldLabel(field) : "";
+  const [label, setLabel] = useState(uidFieldLabel ?? "");
+  const [error, setError] = useState<string | undefined>();
+
+  function handleOpenChange(open: boolean) {
+    setOpen(open);
+    if (!open) {
+      setLabel(uidFieldLabel ?? "");
+      setError(undefined);
+    }
```

Comment:
`handleOpenChange` is not called when you pass `false` to the `open` property of the dialog. That means that in those cases you won't be reseting the form.

To fix this, in concurrent React is ok to do something like this in the render function:
```
if (!open) { setLabel(...), etc }
```
when a set state happens in the render, react interrupt the rendering and starts over with the new values

this makes sure that no mater who changes the open prop, you'll have an opportunity to reset.

Reply:
nice catch, will add that

Reply:
actually, as mentioned in private message, I'll need a bit more help with that